src/formattedcode/writers.py

```diff
@@ -27,235 +27,17 @@
-            file_entry.add_lics(NoAssert())
-
-        else:
-            file_entry.add_lics(SPDXNone())
-
-        file_entry.conc_lics = NoAssert()
-
-        file_copyrights = file_data.get('copyrights')
-        if file_copyrights:
-            all_files_have_no_copyright = False
-            file_entry.copyright = []
-            for file_copyright in file_copyrights:
-                file_entry.copyright.extend(file_copyright.get('statements'))
-
-            doc.package.cr_text.update(file_entry.copyright)
-
-            # Create a text of copyright statements in the order they appear in the file.
-            # Maintaining the order might be useful for provenance purposes.
-            file_entry.copyright = '\n'.join(file_entry.copyright) + '\n'
-
-        elif file_copyrights is None:
-            all_files_have_no_copyright = False
-            file_entry.copyright = NoAssert()
-
-        else:
-            file_entry.copyright = SPDXNone()
-
-
-        doc.package.add_file(file_entry)
-
-    if len(doc.package.files) == 0:
-        if format == 'spdx-tv':
-            output_file.write("# No results for package '{}'.\n".format(doc.package.name))
-        elif format == 'spdx-rdf':
-            output_file.write("<!-- No results for package '{}'. -->\n".format(doc.package.name))
-
-    # Remove duplicate licenses from the list for the package.
-    unique_licenses = set(doc.package.licenses_from_files)
-    if not len(doc.package.licenses_from_files):
-        if all_files_have_no_license:
-            doc.package.licenses_from_files = [SPDXNone()]
-        else:
-            doc.package.licenses_from_files = [NoAssert()]
-    else:
-        # List license identifiers alphabetically for the package.
-        doc.package.licenses_from_files = sorted(unique_licenses, key=lambda x: x.identifier)
-
-    if len(doc.package.cr_text) == 0:
-        if all_files_have_no_copyright:
-            doc.package.cr_text = SPDXNone()
-        else:
-            doc.package.cr_text = NoAssert()
-    else:
-        # Create a text of alphabetically sorted copyright
-        # statements for the package.
-        doc.package.cr_text = '\n'.join(sorted(doc.package.cr_text)) + '\n'
-
-    doc.package.verif_code = doc.package.calc_verif_code()
-    doc.package.license_declared = NoAssert()
-    doc.package.conc_lics = NoAssert()
-
-    if format == 'spdx-tv':
-        from spdx.writers.tagvalue import write_document
-    elif format == 'spdx-rdf':
-        from spdx.writers.rdf import write_document
-
-    # As the spdx-tools package can only write the document to a
-    # "str" file but ScanCode provides a "unicode" file, write to a
-    # "str" buffer first and then manually write the value to a
-    # "unicode" file.
-    from StringIO import StringIO
-    str_buffer = StringIO()
-    write_document(doc, str_buffer, validate=True)
-    as_unicode = str_buffer.getvalue().decode('utf-8')
-    output_file.write(as_unicode)
+    output_plugin.write_output(
```

Comment:
I got how to get the `hookcaller` of a plugin, but without the callable name, how do we differentiate one caller from other?